### PR TITLE
Add single/dual channel texture support to XBT textures

### PR DESCRIPTION
**Pull request by sarbes**
## Description
This PR adds support for single and dual channel textures to the texture packer and the XBT processing path.

The packer will convert the provided textures to a R8/RG8 texture with an related swizzle, if possible.

The XBT loader will pass the texture/swizzle on, so the texture can be directly uploaded to the GPU (GL/GLES 3.0). On systems which don't support such textures (DX, GLES 2.0 only for luminance/uminance+alpha), the texture will be converted to BGRA.

This is a revival of https://github.com/xbmc/xbmc/pull/21557, but from a (slightly) different angle.

## Motivation and context
Including such textures has a lot of advantages:
- less disk space used (-500KB for Estuary)
- smaller package size (-100KB)
- smaller memory footprint (storage/bandwidth) on the GPU if supported (-50%/-75%)
- faster load times

## How has this been tested?
Runs fine under GL/GLES.

## What is the effect on users?
Potentially less jank and more FPS.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by sundermann — COMMENTED** [review]
Tried it on macOS and webOS and both seem ok.

File: xbmc/guilib/TextureBase.cpp [hunk 1]
```diff
@@ -398,3 +398,77 @@ void CTextureBase::SetKDFormat(XB_FMT xbFMT)
...
+  size_t size = GetPitch() * GetRows();
+
+  Allocate(width, height, XB_FMT_A8R8G8B8);
+
+  if (m_textureSwizzle == KD_TEX_SWIZ_111R)
+  {
+    for (int32_t i = size - 1; i >= 0; i--)
+    {
+      m_pixels[i * 4 + 3] = src[i];
+      m_pixels[i * 4 + 2] = char(0xff);
```
[the remark this answers is not in the document]
Comment (sarbes) [reply xbmc/guilib/TextureBase.cpp]: Yeah, you are right.

**Comment by sarbes** [comment]
The Windows version runs fine under wine (both XBT v2 and v3).

@sundermann can I get a green light to merge?

**Review by sundermann — APPROVED** [review]

**Comment by sarbes** [comment]
Ty

**Pull request merged**